prov/efa/src/rdm/rxr_pkt_type_base.c

```diff
@@ -122,24 +123,35 @@ int rxr_pkt_init_data_from_op_entry(struct rxr_ep *ep,
 
 	rxr_locate_iov_pos(op_entry->iov, op_entry->iov_count, tx_data_offset,
 			   &tx_iov_index, &tx_iov_offset);
-	desc = op_entry->desc[0];
+	iov_mr = op_entry->desc[tx_iov_index];
 	assert(tx_iov_index < op_entry->iov_count);
 	assert(tx_iov_offset < op_entry->iov[tx_iov_index].iov_len);
 
-	ret = rxr_ep_use_p2p(ep, desc);
-	if (ret == 0)
-		goto copy;
+	if (pkt_entry->mr) {
+		/* When using EFA device, EFA device can access memory that
+		 * that has been registered, and p2p is allowed to be used.
+		 */
+		if (iov_mr) {
+			ret = rxr_ep_use_p2p(ep, iov_mr);
+			if (ret < 0)
+				return ret;
```

Comment:
I think it's better(easier to debug) if we do `assert(ret >= 0)` instead of return.

Reply:
assertion is for something cannot happen. It can happen when that `ret < 0`. so we need to return and print an error message